dotnet/src/SemanticKernel/Orchestration/SKContextExtensions.cs

```diff
@@ -15,13 +15,19 @@ internal static class SKContextExtensions
     /// <param name="skillName">The skill name</param>
     /// <param name="functionName">The function name</param>
     /// <param name="registeredFunction">The registered function, if found</param>
-    internal static bool IsFunctionRegistered(this SKContext context, string skillName, string functionName, [NotNullWhen(true)] out ISKFunction? registeredFunction)
+    internal static bool IsFunctionRegistered(this SKContext context,
+        string? skillName,
+        string functionName,
+        [NotNullWhen(true)] out ISKFunction? registeredFunction)
     {
```

Comment:
See comments in https://github.com/microsoft/semantic-kernel/pull/849
 - Recommend removing the word "registered" from method and args, since the Register* methods are all now Import* or AddFunction
 - Rename this method to TryGetFunction for consistency with dotnet and SK's other similar methods

Reply:
what about renaming both to `IsFunctionAvailable`? the name would provide a more fluent API

Reply:
> what about renaming both to IsFunctionAvailable? the name would provide a more fluent API

What do you do after such a method returns true?  Isn't the 99% case to then get the function, in which case the simpler and cheaper thing is to just do it all in a single lookup.  If you don't care about the resulting object, the usability downside to `TryGetFunction` is just an additional `, out _` at the call site, but if you do need the object and you have to make a separate call to get it, it means a much larger bit of code needs to be tacked on.

If we're trying to make this feel like a collection of skills, then it should mirror the shape of collections in .NET.  None of them have "IsAvaialable" APIs.  If there are common cases where the queried for function isn't needed and you only need to know whether it's contained or not, we could also expose "ContainsFunction" in addition to "TryGetFunction".

Reply:
Yeah, I agree... I believe, if it returns bool and has out var, then it's a TryGet by convention/expectation. If it just returns a bool (no out var), it can be a Has* or Is*. We could have both methods, but the TryGet* is a superset, and is the better one to have as it's a single lookup for both has+get.

Reply:
IMHO I would have two methods 

1. `GetFunctionOrDefault` or just `GetFunction` (not throwing):
2. `TryGetFunction`

```
bool TryGetFunction (...params, out ISKFunction? x) 
{
    x = GetOrDefault(...params);
    return x is null;
}

ISKFunction? GetOrDefault(...params) 
{ 
    ... 
}
```

Reply:
each language has a specific approach, looking at options across multiple languages

C#: bool TryGet(key, out value)
python: T get(key, default) // no bool
Java: T get(key) // null if not exists
TS:  [key] // undefined is not exists